### ISSUE TITLE
Race Driver 2006, Simulate UMD delays crashes or hangs in Simulator Modes "Now Loading" screens (Windows)

### ISSUE THREAD

**Issue by issuer** [hrydgard/ppsspp#7425]
The game crashes PPSSPP 1.0 with FastMem after the loading screen just before each race in Simulator Modes and hangs with FastMem disabled.
Oddly enough, the game gets past the loading screen and is playable on the Trans World Cup mode and since it's by the same developer of the DTM \ ToCa \ V8 Supercar games (CODEMASTERS) has the same issues as #7294.

Warning spam in said loading screen below. Also, no errors are logged, just warnings.
![rd2006_warning_spam](https://cloud.githubusercontent.com/assets/10441372/6025630/2263b202-abbb-11e4-92c4-437a4e604713.png)

**Comment by unknownbrackets** [comment]
What's the log look like before the first invalid address error?  The actual invalid addresses don't tell us much.

-[Unknown]

**Comment by issuer** [comment]
I'm confus. Tested a couple of times and it seems completely random, sometimes it works sometimes it doesn't. I managed to get two full debug logs.

CRASH: https://www.dropbox.com/s/jxgovpco5zuw3nl/ppsspplog_RD2006_CRASH.rar?dl=0
Got past the "Now Loading" screen due to wizardry: https://www.dropbox.com/s/2x4o39rk1adwdns/ppsspplog_RD2006_WORKED.rar?dl=0

**Comment by daniel229** [comment]
I can't reproduce the invalid addresses.

Here is the artifact from Gedebugger
![01](https://cloud.githubusercontent.com/assets/3481559/6035506/ce1f3c2e-ac6f-11e4-90a8-579efe0dd763.png)
![02](https://cloud.githubusercontent.com/assets/3481559/6035505/ce1b4a2e-ac6f-11e4-8227-a307b6fda0ee.png)
more tabs
https://gist.github.com/daniel229/62e51403df59a762bac1

**Comment by Bigpet** [comment]
do the crashes happen without jit?

**Comment by daniel229** [comment]
I can't reproduce the invalid addresses in Time Trial mode,and it alway happens,I tested many build since 0.8.1.disable jit not help.

**Comment by issuer** [comment]
@Bigpet, yes they do, it must be some setting i'm using since i mess with my settings quite often but i still can't figure it out what it is that is causing it to hang or crash.

I went on and got the latest dev build from orphis v1.0-38-g05fc4cf to test daniel229's reply...
To my surprise the game don't spam invalid addresses anymore and thus doesn't crash, i've just noticed a couple of SCE_KERNEL_NO_ASYNC but i guess that may be fine. 

It's definitely some setting i'm using that's causing this, but i just don't know what can it be. Disabling JIT did not help and doesn't seem related to it. I'll do some more testing and report back when i get these and what's causing them if i do manage to replicate the issue.

**Comment by issuer** [comment]
Wow today must be my lucky day. Found the culprit on the second try.

I/O timing method: Simulate UMD Delays is causing the crashes (with FastMem) and the hangs (without FastMem).

I didn't know changing the I/O timing method could be this harmful. In fact, i believed Simulate UMD Delays was safer than Fast and Host.

**Comment by unknownbrackets** [comment]
Simulate UMD delays is new and not entirely stable.  Currently, it fixes issues in a few games and creates new issues in a few other games.  Even native fixes issues in some games, which is somewhat scary since that's the non-deterministic one.

It's possible that most of the affected issues are related to IO timing in general (not using the game's requested thread priorities to schedule it), or that they're all caused by a couple completely unrelated timing issues outside IO (possibly even the same issue, like even networking.)

-[Unknown]

**Comment by issuer** [comment]
Some more pieces to solve the puzzle in my brain, your explanation was really enlightening.
So, the games request threads to be executed at specific times, if one gets in too early or too late then the whole thing falls apart. I understand now the purpose of the I/O timing.

Also, it is up to you to close this issue or not.
I will not close it since it is indeed a bug with Race Driver 2006 and Simulate UMD delays but it was my fault in the first place changing from the default value Fast that has caused all of this misunderstanding.

**Comment by unknownbrackets** [comment]
Has this improved in the latest git builds?  Some of IO timing has been made more accurate, at the thread level.

I'm hoping that may have improved "Simulate UMD Delays" for this game.

-[Unknown]

**Comment by Panderner** [comment]
> Has this improved in the latest git builds? Some of IO timing has been made more accurate, at the thread level.
> 
> I'm hoping that may have improved "Simulate UMD Delays" for this game.
> 
> -[Unknown]

No. It doesn't improved. Crashes in simulator modes even in fast memory turned off

**Comment by benderscruffy** [comment]
this goes ingame now with Simulate UMD delays
can be closed

**Comment by CookiePLMonster** [comment]
This probably should be re-opened. As of 1.16.6 the game still crashes on loading screens.

**Comment by hrydgard** [comment]
Sure. Though, 16.6 is not the latest version.

**Comment by CookiePLMonster** [comment]
For completeness, I have re-tested on v1.17.1, and the same issue persists.
![image](https://github.com/hrydgard/ppsspp/assets/7947461/f45facf4-f55b-4176-b47f-a5506142f841)

Funnily enough, my initial test was actually incorrect as I had Host timings selected. However, **this test was performed with Simulate UMD Delays, yet the issue manifests itself the same way anyway.**

**Comment by Starlightbotanist** [comment]
I have tested this on currently builds. Still happens regardless of IO timings on many of the races. Some races can be loaded, but not all.

**Comment by yggdrzil** [comment]
I'm having the same problems, although my case is the Career mode. Some championships always kept crashing despite my attempts

Also I play in mobile

**Comment by Maarcis12** [comment]
Tested out the latest build, crashes are gone, i can progress in career mode perfectly fine now

<img width="1920" height="1080" alt="Image" src="https://github.com/user-attachments/assets/9a4d263a-76ae-4019-9d2d-f7e0a7f5b01c" />

<img width="1920" height="1080" alt="Image" src="https://github.com/user-attachments/assets/92490040-9712-4dd8-a51f-94d095f65664" />
